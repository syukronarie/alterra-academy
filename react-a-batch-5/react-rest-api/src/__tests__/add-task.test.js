import { fireEvent, render, screen } from "@testing-library/react";
import AddTask from "../components/task/add-task";

describe("AddTask", () => {
	it("should render correctly without crash", () => {
		const { container } = render(<AddTask />);
		expect(container).toBeInTheDocument();
	});

	it("should match with snapshots", () => {
		const { container } = render(<AddTask />);
		expect(container).toMatchSnapshot();
	});

	it('input element must has placeholder "Add Task"', () => {
		render(<AddTask />);
		const inputElement = screen.getByPlaceholderText(/add task/i);
		expect(inputElement).toHaveAttribute("placeholder", "Add task");
	});

	it('button element must has text "Add"', () => {
		render(<AddTask />);
		const buttonElement = screen.getByText(/add/i);
		expect(buttonElement).toHaveTextContent("Add");
	});

	it("should trigger input onChange correctly", () => {
		render(<AddTask />);
		const inputElement = screen.getByPlaceholderText(/add task/i);
		fireEvent.change(inputElement, { target: { value: "learn unit test" } });
		expect(inputElement).toHaveValue("learn unit test");
	});

	it("should trigger button onClick correctly", () => {
		render(<AddTask />);
		const buttonElement = screen.getByText(/add/i);
		const inputElement = screen.getByPlaceholderText(/add task/i);
		fireEvent.change(inputElement, { target: { value: "learn unit test" } });
		fireEvent.click(buttonElement);
		expect(inputElement).toHaveValue("");
	});
});
