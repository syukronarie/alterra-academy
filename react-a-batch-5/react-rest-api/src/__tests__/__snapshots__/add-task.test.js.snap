// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTask should match with snapshots 1`] = `
<div>
  <input
    placeholder="Add task"
    value=""
  />
  <button>
    Add
  </button>
</div>
`;
